src/qemu.rs

```diff
@@ -98,7 +102,11 @@ fn gen_init() -> Result<NamedTempFile> {
         .set_permissions(perms)
         .context("Failed to set executable bits on init tempfile")?;
 
-    Ok(f)
+    // Path in the guest needs to be stripped from the rootfs path.
+    // This will remove the leading `/` so we need to tuck it back in.
+    let stripped_init = Path::new("/").join(f.path().strip_prefix(rootfs.canonicalize()?)?);
```

Comment:
IIUC this should be equivalent. A bit easier to read IMO

```suggestion
    let stripped_init = Path::new("/").join(f.path().file_name().unwrap();
    let stripped_init = Path::new("/").join(f.path().strip_prefix(rootfs.canonicalize()?)?);
```

Reply:
not really.... say your rootfs directory is:
/a/b/c/rootfs

f will be something like:
/a/b/c/rootfs/tmp/vminit-random.sh

Your suggestion would make stripped_init equal to /vminit-random.sh

while my solution would be:
/tmp/vminit-random.sh



Unittests around that code could be useful :)

Reply:
Ah right. Feel free to add a test for this :)

Reply:
ok, I made something less convoluted. Still have not added tests... mostly because there is the NamedTempFile in the middle of that function that mutate the state of the system and would require the temp directory to exist and be writable.
Before I get to add the tests, I updated the PR to reflect the change. Lemme know how that looks now.